### PR TITLE
#486 - Add git branch to statusbar

### PR DESCRIPTION
**Pull request by Galbar**
* Add git branch indicator to status bar

* Add `getBranch()` to `Git` in `Services` and expose it to plugins through `Oni.services.git`
* Expose ui icon creation to plugins through `Oni.ui.createIcon()`

screenshot from 2017-07-15 13-47-00

### CODE REVIEW

**Comment by bert88sta** [comment]
@Galbar: Thanks for picking up after me! Do you think you could potentially model this after the zsh git integration?

**Comment by Galbar** [comment]
@bert88sta: How do you mean?

**Comment by bert88sta** [comment]
@Galbar: I'm not sure of the syntax exactly, I believe it's  * to indicate dirty, + to indicate commits or something along those lines. I'll get a link for you.

**Comment by bert88sta** [comment]
@Galbar: 
image

File: browser/src/Plugins/Api/Ui.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { Icon, IconProps, IconSize } from "../../UI/Icon"
+
+export class Ui {
+    constructor(private _react: any) {
+    }
+
+    public createIcon(props: IconProps): any {
+        return this._react.createElement(Icon, props)
```
Comment (extr0py) [inline browser/src/Plugins/Api/Ui.ts]: Cool, thanks for including both this and `Oni.services.git` in our API surface. Great idea!

File: browser/src/Services/Git.ts [hunk 2]
```diff
@@ -31,3 +35,20 @@ export function getUntrackedFiles(exclude: string[]): Q.Promise<string[]> {
     const untrackedFiles = execSync(cmd).toString("utf8").split("\n")
     return Q.resolve(untrackedFiles)
 }
+
+export function getBranch(path?: string): Q.Promise<string> {
+        const options: IExecOptions = {}
```
Comment (extr0py) [inline browser/src/Services/Git.ts]: Glad we can share this functionality between the core code and plugins.

**Comment by extr0py** [comment]
Change looks great to me! Thanks @Galbar!

There are a few things we could look at as next steps:
- @bert88sta 's feedback regarding additional status info (ie, files modified/uncomitted/etc) . We might be able to get some of this info from the output of the git 
- Detecting when the branch has changed, since currently we only update on entering a buffer (for example, a user changed branch in the terminal). We could potentially add an event on the Git shared object like `branch-changed` that the plugin could subscribe to.
- A configuration option to enable / disable the statusbar (ie, `git.statusBar.enabled`)

But IMO we can tackle those points as separate issues and bring in the PR as-is, and continue to build on it. This is something I've missed in my workflow so I'm excited to have it back! Nice work! 👍 👍

**Pull request merged**